### DISCUSSION TITLE
Previous state of model in after hook

### DISCUSSION THREAD

**Discussion by iarkaroy** [adonisjs/core#4081] [Help]
Is it possible to get the previous state of the model (like the `$original` property) in the after hook (like `@afterSave`)? I could not find anything like that.

Any guidance would be much appreciated.

**Comment by theRealPadster** [comment]
I know this is really old, but came across it while looking for an answer to another problem I had, and thought I'd share my solution for this one. I added a custom field called "originalForHooks" in my model, and fill it in in the `beforeSave` hook. Then I can use this value in my `afterSave` hook. 

```ts
@beforeSave()
  public static async setPublishedAt(ad: PartAd) {
    // Store the original values in a custom property
    ad.originalForHooks = { ...ad.$original };
    // ...
}
```
